log4j-api/src/main/java/org/apache/logging/log4j/spi/Provider.java

```diff
@@ -271,75 +223,39 @@ public Class<? extends LoggerContextFactory> loadLoggerContextFactory() {
         return null;
     }
 
-    private LoggerContextFactory createLoggerContextFactory() {
-        final Class<? extends LoggerContextFactory> factoryClass = loadLoggerContextFactory();
-        if (factoryClass != null) {
-            try {
-                return LoaderUtil.newInstanceOf(factoryClass);
-            } catch (final Exception e) {
-                LOGGER.error(
-                        "Unable to create instance of class {} specified in {}", factoryClass.getName(), getUrl(), e);
-            }
-        }
-        LOGGER.warn("Falling back to {}", SimpleLoggerContextFactory.INSTANCE);
-        return SimpleLoggerContextFactory.INSTANCE;
-    }
-
     /**
      * @return The logger context factory to be used by {@link org.apache.logging.log4j.LogManager}.
      * @since 2.24.0
      */
     public LoggerContextFactory getLoggerContextFactory() {
-        return loggerContextFactoryLazy.get();
```

Comment:
Any particular reasons we moved away from the `Lazy`? (I support the change, BTW.)

Reply:
This method and `getThreadContextMapInstance()` are called only **once** in `log4j-api`.

`Lazy` is still used in the `Log4jProvider` implementation, so we don't need to access static methods to create a `ContextDataInjector`, for example.

In Log4j Core 3, `Lazy` is replaced by DI scopes.